tests/test_eigen_tensor.cpp

```diff
@@ -0,0 +1,259 @@
+/*
+    tests/eigen_tensor.cpp -- automatic conversion of Eigen Tensor
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#include <pybind11/eigen.h>
+
+#include "pybind11_tests.h"
+
+template <typename M>
+void reset_tensor(M &x) {
+    for (int i = 0; i < 3; i++) {
```

Comment:
Is it not possible to look up these dimensions in the matrix to avoid magic numbers?